File: app/src/main/java/it/unical/mat/embasp/asp/ASPFilterOption.java
package it.unical.mat.embasp.asp;

import it.unical.mat.embasp.base.OptionDescriptor;

/** Generic filter option for ASP solver */

public abstract class ASPFilterOption extends OptionDescriptor {
	public ASPFilterOption() {
		super("-filter=");
	}
}

File: Python version/languages/asp/asp_filter_option.py
from abc import ABCMeta
from base.option_descriptor import OptionDescriptor


class ASPFilterOption(OptionDescriptor):
    """Generic filter option for ASP solver."""

    __metaclass__ = ABCMeta

    def __init__(self):
        super(ASPFilterOption, self).__init__("-filter=")
